stdlib/asm/collections/mmr.masm

```diff
@@ -0,0 +1,81 @@
+#! Computes the `ilog2(number)` and `2^(ilog2(number))`.
+#!
+#! number must be non-zero, otherwise this will error
+#!
+#! Stack transition:
+#! [number, ...] -> [ilog2, power_of_two, ...]
+export.ilog2_checked
+  push.2147483648 # power_of_two from high to low bit
+  push.0          # bit_pos from the most-signficant, `31-bit_pos` equals to ilog2
+  # stack: [bit_pos, power_of_two, number, ...]
+
+  # find the first most-significant true bit
+  dup.1 dup.3 u32checked_and eq.0 while.true
+    add.1 swap div.2 swap
+    dup.1 dup.3 u32checked_and eq.0
+  end
+
+  # compute ilog2
+  push.31 swap sub
+  # stack: [ilog2, power_of_two, number, ...]
+
+  # drop number
+  movup.2 drop
+  # stack: [ilog2, power_of_two, ...]
+end
+
+#! Loads the leaf at the absolute `pos` in the MMR.
+#!
+#! Stack transition:
+#! [pos, mmr_ptr, ...] -> [n3, n2, n1, n0, ...]
```

Comment:
note:

- the initial design was `first_ptr, last_ptr` [ref](https://github.com/0xPolygonMiden/crypto/issues/51#issuecomment-1426828580)
- this was changed to a single element, with both the address and size packed into it, the idea being it was easier for the user to have a single element in the stack instead of two.
- here i moved the size to the memory address, this lifts the constraint of memory addresses that could be used to store an MMR that the packed handle would have, still is convenient to the user, and from my rough cycle count, is more efficient (loading from memory is just two cycles, the pack/unpacking would be around 3)